native/native_main.py

```diff
@@ -118,6 +130,207 @@ def sanitizeFilename(fn):
     return fn
 
 
+def is_valid_firefox_profile(profile_dir):
+    is_valid = False
+    validity_indicator = "times.json"
+
+    if pathlib.WindowsPath(profile_dir).is_dir():
+        test_path = "%s\\%s" % (profile_dir, validity_indicator)
+
+        if pathlib.WindowsPath(test_path).is_file():
+            is_valid = True
+
+    return is_valid
+
+
+def win_firefox_restart(message):
+    """Handle 'win_firefox_restart' message."""
+    reply = {}
+    profile_dir = message["profiledir"].strip()
+
+    ff_bin_name = "firefox.exe"
+    ff_lock_name = "parent.lock"
+
+    if profile_dir != "auto" \
+            and not is_valid_firefox_profile(profile_dir):
+        reply = {
+            "cmd": "error",
+            "error": "%s %s %s" % (
+                "Invalid profile directory specified.",
+                "Vaild profile directory path(s) can be found by",
+                "navigating to 'about:support'."
+            )
+        }
+
+    elif not is_command_on_path(ff_bin_name):
+        reply = {
+            "cmd": "error",
+            "error": "firefox.exe is not found on %PATH%."
+        }
+
+    else:
+        # {{{
+        # Native messenger can't seem to create detached process on
+        # Windows while Firefox is quitting, which is essential to
+        # trigger restarting Firefox. So, below we are resorting to
+        # create a scheduled task with the task start-time set in
+        # the near future.
+        #
+
+        #
+        # subprocess.Popen(
+        #    [ff_bin_path, "-profile", profile_dir],
+        #    shell=False,
+        #    creationflags=0x208 \
+        #    | subprocess.CREATE_NEW_PROCESS_GROUP)
+        #
+
+        #
+        # 'schtasks.exe' is limited as in it doesn't support
+        # task-time with granularity in seconds. So, falling back
+        # to PowerShell as the last resort.
+        #
+
+        # out_str = ""
+        # task_time = time.strftime("%H:%M",
+        #                           time.localtime(
+        #                               time.time() + 60))
+        #
+        # out_str = subprocess.check_output(
+        #     ["schtasks.exe",
+        #      "/Create",
+        #      "/F",
+        #      "/SC",
+        #      "ONCE",
+        #      "/TN",
+        #      "tridactyl",
+        #      "/TR",
+        #      "calc",
+        #      "/IT",
+        #      "/ST",
+        #      task_time],
+        #     shell=True)
+        # }}}
+
+        ff_bin_path = "\"%s\"" % shutil.which(ff_bin_name)
+        new_tab_url = "https://www.mozilla.org/en-US/firefox/"
+
+        if profile_dir == "auto":
+            ff_lock_path = ff_bin_path
+            ff_args = "\"%s\" \"%s\" \"%s\" \"%s\"" % \
+                ("-foreground",
+                 "-ProfileManager",
```

Comment:
Why do you have ProfileManager? Most people just use one profile, so seeing this appear will be a bit confusing. Is it because you have no way of knowing which profile is being used? I think I would rather not have it, as firefox will show it automatically if someone has not selected a default, which is the likely case if someone is using multiple profiles.

Reply:
[Done][0].

[0]: https://github.com/gsbabil/tridactyl/commit/b6b71c44678cee08b3e6a7b8472a15c82b9e0ebf